src/frontend/src/binder/expr/function.rs

```diff
@@ -170,6 +170,19 @@ impl Binder {
             }
         }
 
+        // datatype cast function
+        // only functions required by the existing PostgreSQL tool are implemented
+        #[allow(clippy::single_match)] // remove it when more cast functions needed
+        match (function_name.as_str(), inputs.as_slice()) {
+            ("date", [expr]) => {
+                return expr
+                    .clone()
+                    .cast_explicit(DataType::Date)
+                    .map_err(Into::into);
+            }
```

Comment:
It can just be another builtin scalar function in this gigantic table, handled by `guard_by_len` and `raw`.
https://github.com/risingwavelabs/risingwave/blob/d2e37ea5ffc6d85f8a62c8abf492e7456f6e6f4e/src/frontend/src/binder/expr/function.rs#L675

Reply:
Is this okay now?